src/main.rs

```diff
@@ -258,12 +281,21 @@ impl Blog {
 
     fn render_feed(&self) -> Result<(), Box<Error>> {
         let posts: Vec<_> = self.posts.iter().by_ref().take(10).collect();
-        let data = json!({ "posts": posts, "feed_updated":  time::now_utc().rfc3339().to_string() });
+        let data =
+            json!({ "posts": posts, "feed_updated":  time::now_utc().rfc3339().to_string() });
 
         self.render_template("feed.xml", "feed", data)?;
         Ok(())
     }
 
+    fn generate_releases_feed(&self) -> Result<(), Box<Error>> {
+        let posts: Vec<_> = self.posts.iter().filter(|post| post.release).collect();
+        let data =
+            json!({"releases": posts, "feed_updated": time::now_utc().rfc3339().to_string()});
```

Comment:
It would probably be worth making an actual struct and deriving serialize on it.

Reply:
Do you want it to be 
```rust
struct Releases {
    releases: Vec<Post>,
    fead_updated: Time
}
```

or something else/with more?

Reply:
Yep, that's perfect - I prefer having the struct to inline json macro calls.

Reply:
I'm still unsure how to you actually want to write this to a file.

Reply:
Something along the lines of the following. Does that make sense?

```rust
fs::write(self.directory.join("releases.json"), serde_json::to_string(&json)?)?
```